astroid/tests/unittest_inference.py

```diff
@@ -2218,6 +2218,31 @@ def no_yield_mgr():
         self.assertRaises(InferenceError, next, module['other_decorators'].infer())
         self.assertRaises(InferenceError, next, module['no_yield'].infer())
 
+    def test_nested_contextmanager(self):
+        """Make sure contextmanager works with nested functions
+
+        Previously contextmanager would retrieve
+        the first yield instead of the yield in the
+        proper scope
+
+        Fixes https://github.com/PyCQA/pylint/issues/1746
+        """
+        code = """
+        from contextlib import contextmanager
+
+        @contextmanager
+        def outer():
+            @contextmanager
+            def inner():
+                yield 2
+            yield inner
+
+        with outer() as foo:
+            foo #@
+        """
+        value = next(extract_node(code).infer())
```

Comment:
Maybe we should also do an inference test, where we'd infer the value returned by the inner context manager, something along the lines of:
```
n = extract_node('''
import contextlib
@contextlib.contextmanager
def test():
  @contextlib.contextmanager
  def inner():
      yield 2
  yield inner

with test() as ctx:
   with ctx as y:
       y #@
''')
assert isinstance(next(n.infer()), Const)
```

Reply:
Good idea